Despite what Travis-CI says, I can't get this project to build

At 1f53c45faf8feb18500f83e031516a14e66ec8d9, `mvn clean verify` fails with:

```
[ERROR] Failed to execute goal org.eclipse.tycho:tycho-compiler-plugin:0.26.0:compile (default-compile) on project org.jboss.tools.vscode.java: Compilation failure: Compilation failure:
[ERROR] /Users/mbolin/src/java-language-server/org.jboss.tools.vscode.java/src/org/jboss/tools/vscode/java/internal/handlers/HoverHandler.java:[37]
[ERROR] $.setContents(Arrays.asList(hover));
[ERROR] ^^^^^^^^^^^
[ERROR] The method setContents(Either<Either<String,MarkedString>,List<Either<String,MarkedString>>>) in the type Hover is not applicable for the arguments (List<String>)
[ERROR] /Users/mbolin/src/java-language-server/org.jboss.tools.vscode.java/src/org/jboss/tools/vscode/java/internal/handlers/HoverHandler.java:[37]
[ERROR] $.setContents(Arrays.asList(hover));
[ERROR] ^^^^^^^^^^^^^^^^^^^^
[ERROR] Type mismatch: cannot convert from List<String> to Either<Either<String,MarkedString>,List<Either<String,MarkedString>>>
[ERROR] /Users/mbolin/src/java-language-server/org.jboss.tools.vscode.java/src/org/jboss/tools/vscode/java/internal/handlers/JDTLanguageServer.java:[181]
[ERROR] public CompletableFuture<CompletionList> completion(TextDocumentPositionParams position) {
[ERROR] ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[ERROR] The return type is incompatible with TextDocumentService.completion(TextDocumentPositionParams)
[ERROR] /Users/mbolin/src/java-language-server/org.jboss.tools.vscode.java/src/org/jboss/tools/vscode/java/internal/handlers/JDTLanguageServer.java:[221]
[ERROR] public CompletableFuture<List<? extends Location>> definition(TextDocumentPositionParams position) {
[ERROR] ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[ERROR] The return type is incompatible with TextDocumentService.definition(TextDocumentPositionParams)
[ERROR] 4 problems (4 errors)
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR]
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :org.jboss.tools.vscode.java
```

--- a/org.jboss.tools.vscode.java/src/org/jboss/tools/vscode/java/internal/handlers/CompletionHandler.java
+++ b/org.jboss.tools.vscode.java/src/org/jboss/tools/vscode/java/internal/handlers/CompletionHandler.java
@@ -23,6 +23,7 @@ import org.eclipse.lsp4j.CompletionItem;
 import org.eclipse.lsp4j.CompletionList;
 import org.eclipse.lsp4j.TextDocumentPositionParams;
 import org.eclipse.lsp4j.jsonrpc.CompletableFutures;
+import org.eclipse.lsp4j.jsonrpc.messages.Either;
 import org.jboss.tools.vscode.java.internal.CancellableProgressMonitor;
 import org.jboss.tools.vscode.java.internal.JDTUtils;
 import org.jboss.tools.vscode.java.internal.JavaLanguageServerPlugin;
@@ -30,7 +31,7 @@ import org.jboss.tools.vscode.java.internal.contentassist.CompletionProposalRequ
 
 public class CompletionHandler{
 
-	CompletableFuture<CompletionList> completion(TextDocumentPositionParams position){
+	CompletableFuture<Either<List<CompletionItem>, CompletionList>> completion(TextDocumentPositionParams position){
 		return CompletableFutures.computeAsync(cancelChecker->{
 			ICompilationUnit unit = JDTUtils.resolveCompilationUnit(position.getTextDocument().getUri());
 			List<CompletionItem> completionItems = this.computeContentAssist(unit,
@@ -39,7 +40,7 @@ public class CompletionHandler{
 			CompletionList $ = new CompletionList();
 			$.setItems(completionItems);
 			JavaLanguageServerPlugin.logInfo("Completion request completed");
-			return $;
+			return Either.forRight($);
 		});
 	}
 

--- a/org.jboss.tools.vscode.java/src/org/jboss/tools/vscode/java/internal/handlers/HoverHandler.java
+++ b/org.jboss.tools.vscode.java/src/org/jboss/tools/vscode/java/internal/handlers/HoverHandler.java
@@ -10,7 +10,6 @@
  *******************************************************************************/
 package org.jboss.tools.vscode.java.internal.handlers;
 
-import java.util.Arrays;
 import java.util.concurrent.CompletableFuture;
 
 import org.eclipse.jdt.core.ITypeRoot;
@@ -34,7 +33,7 @@ public class HoverHandler{
 			}
 			Hover $ = new Hover();
 			if (hover != null && !hover.isEmpty()) {
-				$.setContents(Arrays.asList(hover));
+				$.setContents(hover);
 			}
 			return $;
 		});

--- a/org.jboss.tools.vscode.java/src/org/jboss/tools/vscode/java/internal/handlers/JDTLanguageServer.java
+++ b/org.jboss.tools.vscode.java/src/org/jboss/tools/vscode/java/internal/handlers/JDTLanguageServer.java
@@ -47,6 +47,7 @@ import org.eclipse.lsp4j.TextDocumentPositionParams;
 import org.eclipse.lsp4j.TextEdit;
 import org.eclipse.lsp4j.WorkspaceEdit;
 import org.eclipse.lsp4j.WorkspaceSymbolParams;
+import org.eclipse.lsp4j.jsonrpc.messages.Either;
 import org.eclipse.lsp4j.jsonrpc.services.JsonDelegate;
 import org.eclipse.lsp4j.services.LanguageServer;
 import org.eclipse.lsp4j.services.TextDocumentService;
@@ -178,7 +179,7 @@ public class JDTLanguageServer implements LanguageServer, TextDocumentService, W
 	 * @see org.eclipse.lsp4j.services.TextDocumentService#completion(org.eclipse.lsp4j.TextDocumentPositionParams)
 	 */
 	@Override
-	public CompletableFuture<CompletionList> completion(TextDocumentPositionParams position) {
+	public CompletableFuture<Either<List<CompletionItem>, CompletionList>> completion(TextDocumentPositionParams position) {
 		logInfo(">> document/completion");
 		CompletionHandler handler = new CompletionHandler();
 		return handler.completion(position);
@@ -218,7 +219,7 @@ public class JDTLanguageServer implements LanguageServer, TextDocumentService, W
 	 * @see org.eclipse.lsp4j.services.TextDocumentService#definition(org.eclipse.lsp4j.TextDocumentPositionParams)
 	 */
 	@Override
-	public CompletableFuture<List<? extends Location>> definition(TextDocumentPositionParams position) {
+	public CompletableFuture<Either<Location, List<? extends Location>>> definition(TextDocumentPositionParams position) {
 		logInfo(">> document/definition");
 		NavigateToDefinitionHandler handler = new NavigateToDefinitionHandler();
 		return handler.definition(position);

--- a/org.jboss.tools.vscode.java/src/org/jboss/tools/vscode/java/internal/handlers/NavigateToDefinitionHandler.java
+++ b/org.jboss.tools.vscode.java/src/org/jboss/tools/vscode/java/internal/handlers/NavigateToDefinitionHandler.java
@@ -22,14 +22,15 @@ import org.eclipse.jdt.core.JavaModelException;
 import org.eclipse.lsp4j.Location;
 import org.eclipse.lsp4j.Range;
 import org.eclipse.lsp4j.TextDocumentPositionParams;
+import org.eclipse.lsp4j.jsonrpc.messages.Either;
 import org.jboss.tools.vscode.java.internal.JDTUtils;
 import org.jboss.tools.vscode.java.internal.JavaLanguageServerPlugin;
 
 public class NavigateToDefinitionHandler {
 
-	public CompletableFuture<List<? extends Location>> definition(TextDocumentPositionParams position){
+	public CompletableFuture<Either<Location, List<? extends Location>>> definition(TextDocumentPositionParams position){
 		return CompletableFuture.supplyAsync(()->{
-			return getDefinition(position);
+			return Either.forRight(getDefinition(position));
 		});
 	}
 

--- a/org.jboss.tools.vscode.tests/src/org/jboss/tools/vscode/java/internal/handlers/CompletionHandlerTest.java
+++ b/org.jboss.tools.vscode.tests/src/org/jboss/tools/vscode/java/internal/handlers/CompletionHandlerTest.java
@@ -75,7 +75,7 @@ public class CompletionHandlerTest extends AbstractCompletionBasedTest {
 						"	}\n"+
 				"}\n");
 		int[] loc = findCompletionLocation(unit, "Objec");
-		CompletionList list = server.completion(JsonMessageHelper.getParams(createCompletionRequest(unit, loc[0], loc[1]))).join();
+		CompletionList list = server.completion(JsonMessageHelper.getParams(createCompletionRequest(unit, loc[0], loc[1]))).join().getRight();
 		assertNotNull(list);
 		assertFalse("No proposals were found",list.getItems().isEmpty());
 		for ( CompletionItem item : list.getItems()) {
@@ -107,7 +107,7 @@ public class CompletionHandlerTest extends AbstractCompletionBasedTest {
 
 		int[] loc = findCompletionLocation(unit, "java.sq");
 
-		CompletionList list = server.completion(JsonMessageHelper.getParams(createCompletionRequest(unit, loc[0], loc[1]))).join();
+		CompletionList list = server.completion(JsonMessageHelper.getParams(createCompletionRequest(unit, loc[0], loc[1]))).join().getRight();
 
 		assertNotNull(list);
 		assertEquals(1, list.getItems().size());
@@ -148,7 +148,7 @@ public class CompletionHandlerTest extends AbstractCompletionBasedTest {
 
 		int[] loc = findCompletionLocation(unit, "map.pu");
 
-		CompletionList list = server.completion(JsonMessageHelper.getParams(createCompletionRequest(unit, loc[0], loc[1]))).join();
+		CompletionList list = server.completion(JsonMessageHelper.getParams(createCompletionRequest(unit, loc[0], loc[1]))).join().getRight();
 		assertNotNull(list);
 		CompletionItem ci = list.getItems().stream()
 				.filter( item->  item.getLabel().matches("put\\(String \\w+, String \\w+\\) : String"))
@@ -186,7 +186,7 @@ public class CompletionHandlerTest extends AbstractCompletionBasedTest {
 
 		int[] loc = findCompletionLocation(unit, "java.util.Ma");
 
-		CompletionList list = server.completion(JsonMessageHelper.getParams(createCompletionRequest(unit, loc[0], loc[1]))).join();
+		CompletionList list = server.completion(JsonMessageHelper.getParams(createCompletionRequest(unit, loc[0], loc[1]))).join().getRight();
 
 		assertNotNull(list);
 		assertEquals(1, list.getItems().size());

--- a/org.jboss.tools.vscode.tests/src/org/jboss/tools/vscode/java/internal/handlers/HoverHandlerTest.java
+++ b/org.jboss.tools.vscode.tests/src/org/jboss/tools/vscode/java/internal/handlers/HoverHandlerTest.java
@@ -13,6 +13,7 @@ package org.jboss.tools.vscode.java.internal.handlers;
 import static org.jboss.tools.vscode.java.internal.JsonMessageHelper.getParams;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
 
 import java.net.URI;
 import java.nio.file.Paths;
@@ -70,8 +71,8 @@ public class HoverHandlerTest extends AbstractProjectsManagerBasedTest {
 
 		//then
 		assertNotNull(hover);
-		assertEquals("Couldn't find hover for "+payload, 1, hover.getContents().size());
-		assertEquals("Unexpected hover "+hover.getContents().get(0), "This is foo", hover.getContents().get(0));
+		String result = hover.getContents().getLeft().getLeft();//wow this is so elegant!
+		assertEquals("Unexpected hover "+result, "This is foo", result);
 	}
 
 	@Test
@@ -87,8 +88,8 @@ public class HoverHandlerTest extends AbstractProjectsManagerBasedTest {
 
 		//then
 		assertNotNull(hover);
-		assertEquals("Couldn't find hover for "+payload, 1, hover.getContents().size());
-		assertEquals("Unexpected hover "+hover.getContents().get(0), "This is foo", hover.getContents().get(0));
+		String result = hover.getContents().getLeft().getLeft();//wow this is so elegant!
+		assertEquals("Unexpected hover "+result, "This is foo", result);
 	}
 
 	@Test
@@ -104,7 +105,7 @@ public class HoverHandlerTest extends AbstractProjectsManagerBasedTest {
 
 		//then
 		assertNotNull(hover);
-		assertEquals("Shouldn't find hover for "+payload, 0, hover.getContents().size());
+		assertNull("Shouldn't find hover for "+payload, hover.getContents());
 	}
 
 	String createHoverRequest(String file, int line, int kar) {
@@ -131,7 +132,7 @@ public class HoverHandlerTest extends AbstractProjectsManagerBasedTest {
 
 		//then
 		assertNotNull(hover);
-		assertEquals("Couldn't find hover for "+ argParam, 1, hover.getContents().size());
-		assertEquals("Unexpected hover "+hover.getContents().get(0), "String[] args - java.Foo.main(String[])", hover.getContents().get(0));
+		String result = hover.getContents().getLeft().getLeft();//wow this is so elegant!
+		assertEquals("Unexpected hover "+result, "String[] args - java.Foo.main(String[])", result);
 	}
 }

